### ISSUE TITLE
Question About SASRec Performance on Music Dataset and SLMRec Experiment Setup

### ISSUE THREAD

**Issue by Jane-2808140** [WujiangXu/SLMRec#5]
Thank you for sharing your code and datasets for the SLMRec model. I have been trying to reproduce the experiments using your implementation and would like to clarify a few points.

I used the provided train_sr_trad.py script with the preprocessed Music Dataset and only modified the number of epochs to 300 (keeping all other hyperparameters as default). The training and validation results for the final epochs are as follows:
-----------------------------------------------------------------------
Epoch 298 Train Loss: 4.6903
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 230/230 [00:08<00:00, 26.41it/s]
Epoch 298 Val NDCG@10: 0.1567 Best: 0.1567 at epoch 298
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 230/230 [00:08<00:00, 26.64it/s]
Epoch 299 Train Loss: 4.6964
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 230/230 [00:09<00:00, 24.49it/s]
Epoch 299 Val NDCG@10: 0.1574 Best: 0.1574 at epoch 299
Testing best model...
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 230/230 [00:08<00:00, 25.68it/s]
Test metrics: (3.7694104495255845, 0.07968927441596367, np.float64(0.07968927441596367), 0.15601112981289206, np.float64(0.11897327486910304), 0.20817448085658663, np.float64(0.13574217990996945), np.float64(0.12572183231580336))
-------------------------------------------------------------------------
（The test metrics are: stats.loss, stats.hit_1, stats.ndcg_1, stats.hit_5, stats.ndcg_5, stats.hit_10, stats.ndcg_10, stats.MRR）

The reported results in Table 3 of your paper is significantly lower than the results I obtained. Could you clarify if there might be a misunderstanding in the setup (e.g., hyperparameters, data splits, or training configurations)?

If my results are correct, were the SLMRec  trained using embeddings from a SASRec model that achieved better performance will perform better than traditional rec method？

Any guidance you can provide would be greatly appreciated. Thank you for your time!

**Comment by WujiangXu** [comment]
> Thank you for sharing your code and datasets for the SLMRec model. I have been trying to reproduce the experiments using your implementation and would like to clarify a few points.
> 
> ## I used the provided train_sr_trad.py script with the preprocessed Music Dataset and only modified the number of epochs to 300 (keeping all other hyperparameters as default). The training and validation results for the final epochs are as follows:
> ## Epoch 298 Train Loss: 4.6903
> 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 230/230 [00:08<00:00, 26.41it/s]
> Epoch 298 Val NDCG@10: 0.1567 Best: 0.1567 at epoch 298
> 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 230/230 [00:08<00:00, 26.64it/s]
> Epoch 299 Train Loss: 4.6964
> 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 230/230 [00:09<00:00, 24.49it/s]
> Epoch 299 Val NDCG@10: 0.1574 Best: 0.1574 at epoch 299
> Testing best model...
> 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 230/230 [00:08<00:00, 25.68it/s]
> Test metrics: (3.7694104495255845, 0.07968927441596367, np.float64(0.07968927441596367), 0.15601112981289206, np.float64(0.11897327486910304), 0.20817448085658663, np.float64(0.13574217990996945), np.float64(0.12572183231580336))
> （The test metrics are: stats.loss, stats.hit_1, stats.ndcg_1, stats.hit_5, stats.ndcg_5, stats.hit_10, stats.ndcg_10, stats.MRR）
> 
> The reported results in Table 3 of your paper is significantly lower than the results I obtained. Could you clarify if there might be a misunderstanding in the setup (e.g., hyperparameters, data splits, or training configurations)?
> 
> If my results are correct, were the SLMRec trained using embeddings from a SASRec model that achieved better performance will perform better than traditional rec method？
> 
> Any guidance you can provide would be greatly appreciated. Thank you for your time!

Hi, Thank you for your attention. Our method (SLMRec) can indeed achieve better results by using a better embedding layer. The performance differences with SASRec in Table 3 are likely due to different hyperparameter settings. In our experiments, we maintained consistent hyperparameters across multiple datasets.
When using embeddings from a better-performing SASRec model, SLMRec naturally performs better because these embeddings provide richer collaborative filtering information.

**Comment by Jane-2808140** [comment]
Thank you for your response! Additionally, I noticed that the code in finetune.py within the run_finetune.sh script does not seem to include the process of training the teacher model and only contains prediction logic. Did I miss any steps in the workflow?
